moz_ingest/io_modules/decoders/moz_ingest/common.lua

```diff
@@ -164,18 +180,32 @@ local hour = floor(os.time() / 3600)
 
 local function get_geo_city(ip_addr)
     local city = UNK_GEO
+    local subdivision1 = UNK_GEO
+    local subdivision2 = UNK_GEO
     local country = UNK_GEO
 
     local ok, ip = pcall(city_db.lookup, city_db, ip_addr)
-    if not ok then return city, country end
+    if not ok then return city, subdivision1, subdivision2, country end
 
     ok, city = pcall(ip.get, ip, "city", "names", "en")
     if not ok then city = UNK_GEO end
 
+    if city_size_set then
+        -- geoname_id must be in city_size_set
+        local ok, geoname = pcall(ip.get, ip, "city", "geoname_id")
+        if not ok or not city_size_set[geoname] then city = UNK_GEO end
+    end
+
+    -- subdivisions has at most two items
+    ok, subdivision1 = pcall(ip.get, ip, "subdivisions", 0, "iso_code")
+    if not ok then subdivision1 = UNK_GEO end
+    ok, subdivision2 = pcall(ip.get, ip, "subdivisions", 1, "iso_code")
+    if not ok then subdivision2 = UNK_GEO end
```

Comment:
It seems like unavailable sub divisions should just be nil and the field itself should be optional.

Reply:
made sub divisions default to nil, and made reporting them optional as per https://github.com/mozilla-services/lua_sandbox_extensions/pull/285#pullrequestreview-107684022